### PR TITLE
Numberwang, now with more numbers! 

### PR DESCRIPTION
**Pull request by dalpd**
Resolving [#2]
a4BBn6y32SZ8

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -65,5 +70,47 @@ numberWang = do
     putStrLn "Aah, would that it were NumberWang. Alas, it is not."
   numberWang
 
+toRevDigits :: Int -> [Int]
```
Comment (yigitozkavci) [inline app/Main.hs:89]: Would you be able to add comments to every function here please?

File: app/Main.hs [hunk 2]
```diff
@@ -65,5 +70,47 @@ numberWang = do
...
+countInJapanese x =
+  if searchNum x == Nothing
+    then reverse . drop 5 . reverse <$> genNumbers x
+  else searchNum x 
+
+genNumbers :: Int -> Maybe String
+genNumbers x = foldl look Nothing $ zip [0..] (toRevDigits x)
+    where
+      look :: Maybe String -> (Integer, Int) -> Maybe String
+      look acc (1, 1) = searchNum 10 <> Just " " <> acc
```
Comment (yigitozkavci) [inline app/Main.hs:121]: Can you please wrap these `String`, `Int` and `Integer` types with newtypes? Iit's hard to follow otherwise, especially with the presence of folds.

File: app/Main.hs [hunk 3]
```diff
@@ -65,5 +70,47 @@ numberWang = do
...
+      look acc (3, 0) = searchNum 0 <> Just " " <> acc
+      look acc (2, 0) = searchNum 0 <> Just " " <> acc
+      look acc (1, 0) = searchNum 0 <> Just " " <> acc
+
+      look acc (i, v) = searchNum v <> searchNum (10 ^ i) <> Just " " <> acc
+
+digitValues :: Int -> [Int]
+digitValues x = foldl arith [] $ zip [0..] (toRevDigits x)
+    where
+      arith :: [Int] -> (Int, Int) -> [Int]
```
Comment (yigitozkavci) [inline app/Main.hs:141]: Comment these ones too please.

File: app/Tables.hs [hunk 4]
```diff
@@ -100,16 +100,16 @@ baseNumbers =
   [ (1, "ichi")
   , (2, "ni")
   , (3, "san")
-  , (4, "shi") -- yon
+  , (4, "yon")
```
Comment (yigitozkavci) [inline app/Tables.hs:103]: Maybe we should start thinking about how to accept both representations in this case. I'm learning it as "yon" but someone else might be using "shi". Maybe open a separate issue as feature request for this?

File: app/Main.hs [hunk 5]
```diff
@@ -65,5 +70,47 @@ numberWang = do
     putStrLn "Aah, would that it were NumberWang. Alas, it is not."
```
Comment (yigitozkavci) [inline app/Main.hs]: I think we should also show the correct value here in the error message. What do you think? Would it help learners?

**Review by yigitozkavci — CHANGES_REQUESTED** [review]

File: app/Tables.hs (hunk 4 above)
Comment (dalpd) [reply app/Tables.hs:103]: Will do!

File: app/Main.hs (hunk 5 above)
Comment (dalpd) [reply app/Main.hs]: I actually had the correct translation displayed before but decided to take it out since `genNumbers` wasn't implemented yet. I think I can add an option to the CLI, once I start working on it, asking the user whether they want the see answers or not and leave this as is for now.

Comment (yigitozkavci) [reply app/Main.hs]: Asking user for this sounds good!

Comment (yigitozkavci) [reply app/Main.hs]: You can open an issue to make this a feature request, if you wish.

File: app/Main.hs [hunk 6]
```diff
@@ -51,11 +54,16 @@ randomNums x = do
...
+  countInJapanese y
 
+-- `numberWang` generates a digit number (1 to 5) and
+-- using that a random quiz number.
+-- If the guess is equal to the actual translation THAT'S NUMBERWANG!
 numberWang :: IO ()
 numberWang = do
-  num <- randomNums 1
+  g <- newStdGen
+  let h = take 1 (randomRs ('1', '5') g)
```
Comment (yigitozkavci) [inline app/Main.hs]: I think this "1" should be an argument of numberWang. In it's current state it's not possible to work with numbers in range, say 1-1000.

**Review by yigitozkavci — CHANGES_REQUESTED** [review]

File: app/Main.hs [hunk 7]
```diff
@@ -51,19 +54,94 @@ randomNums x = do
...
+  countInJapanese y
 
-numberWang :: IO ()
-numberWang = do
-  num <- randomNums 1
+-- `numberWang` generates a digit number (1 to 5) and
+-- using that a random quiz number.
+-- If the guess is equal to the actual translation THAT'S NUMBERWANG!
+numberWang :: Int -> IO ()
+numberWang x = do
```
Comment (yigitozkavci) [inline app/Main.hs:63]: Up to which digit do we handle in `numberWang`? We should probably warn user if they enter a larger digit count than we can handle.

**Review by yigitozkavci — CHANGES_REQUESTED** [review]

File: app/Main.hs [hunk 8]
```diff
@@ -63,15 +63,19 @@ numberWang :: Int -> IO ()
 numberWang x = do
   g <- newStdGen
   let [h] = take 1 (randomRs (1, x) g)
```
Comment (yigitozkavci) [inline app/Main.hs]: Please prefer non-partial pattern matching here. Like the following:

```haskell
let (h:_) = randomRs (1, x) g
```

**Review by yigitozkavci — CHANGES_REQUESTED** [review]
